### DISCUSSION TITLE
alist无法通过https访问

### DISCUSSION THREAD

**Discussion by shing-yu** [AlistGo/alist#3895] [Q&A]
我在自己的香橙派3lts上部署了alist，并计划通过https进行外网访问，但是出现了一些问题。
http状态码：**502**
我认为可能的原因（请大佬帮我确定）：
1.**使用了ddns-go实现ddns（v6），并用ddns域名申请了ssl证书**
2.家宽**443端口**被限制（基本排除）

以下是config.json配置文件（域名用xxx代替）：
```javascript
{
  "force": false,
  "address": "0.0.0.0",
  "port": 443,
  "site_url": "https://xxx.site",
  "cdn": "",
  "jwt_secret": "MYrfcwGKe5",
  "token_expires_in": 48,
  "database": {
    "type": "sqlite3",
    "host": "",
    "port": 0,
    "user": "",
    "password": "",
    "name": "",
    "db_file": "data/data.db",
    "table_prefix": "x_",
    "ssl_mode": ""
  },
  "scheme": {
    "https": true,
    "cert_file": "/opt/alist/xxx_bundle.crt",
    "key_file": "/opt/alist/xxx.key"
  },
  "temp_dir": "data/temp",
  "bleve_dir": "data/bleve",
  "log": {
    "enable": true,
    "name": "data/log/log.log",
    "max_size": 10,
    "max_backups": 5,
    "max_age": 28,
    "compress": false
  },
  "max_connections": 0,
  "tls_insecure_skip_verify": true
}
``` 

我也尝试了使用nginx进行反向代理，仍然出现无法访问
但nginx**可进行http访问**

**尝试使用宝塔面板进行反向代理，成功！**

恳请各位大佬帮我确定问题，如果能给出解决方案就再好不过了！
( •̀ ω •́ )✧

**Comment by kimycai** [comment]
依据文档说明，证书文件需要放在data目录下才会被识别，所有应该可能是“/opt/alist/data/”下

**Comment by baihy99** [comment]
如果是直接是nginx反向代理https进行域名访问的话，按官方教程在/etc/nginx/nginx.conf添加如下配置即可

> location / {
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header Range $http_range;
	    proxy_set_header If-Range $http_if_range;
      proxy_redirect off;
      proxy_pass http://127.0.0.1:5244;
      # the max size of file to upload
      client_max_body_size 20000m;
}

如果是通过子路径访问的话，如abc.com/alist/，则在/etc/nginx/nginx.conf中添加如下

> location /alist/ {
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header Range $http_range;
      proxy_set_header If-Range $http_if_range;
      proxy_redirect off;
      proxy_pass http://127.0.0.1:5244/alist/;
      # the max size of file to upload
      client_max_body_size 20000m;
}

同时，在/opt/alist/data/config.json修改alist配置文件
修改 "site_url"为“alist”即可
